pretyping/cbv.ml

```diff
@@ -397,6 +397,10 @@ and apply_env env t =
   | _ ->
     map_with_binders subs_lift apply_env env t
 
+let rec strip_app = function
+  | APP (args,st) -> APP (args,strip_app st)
+  | s -> TOP
```

Comment:
```suggestion
  | CASE _ | PROJ _ | TOP -> TOP
```
also the comment on cbv_stack says
~~~
 * Important remark: the APPs should be collapsed:
 *    (APP (l,(APP ...))) forbidden
~~~
so maybe we shouldn't be recursive and just do return `APP (args, TOP)` in the APP case

Reply:
Nothing forces this statically so I think it's a bad idea not to be recursive.